core/src/impl/Kokkos_ConcurrentBitset.hpp

```diff
@@ -138,15 +138,15 @@ struct concurrent_bitset {
     // when is full at the atomic_fetch_add(+1)
     // then a release occurs before the atomic_fetch_add(-1).
 
-    const uint32_t state =
-        (uint32_t)Kokkos::atomic_fetch_add((volatile int *)buffer, 1);
+    const uint32_t state = (uint32_t)Kokkos::atomic_fetch_add(
```

Comment:
The `(uint32_t)` is still a C-style cast (in multiple places in the code).

Reply:
Sure. I am happy to enforce `https://clang.llvm.org/extra/clang-tidy/checks/google-readability-casting.html` but I thought starting with something less controversial.